src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingSearchQuery.cs

```diff
@@ -35,18 +35,28 @@ internal sealed class UnitTestingSearchQuery
         [DataMember(Order = 3)]
         public readonly int MethodParameterCount;
 
-        public static UnitTestingSearchQuery ForType(string fullyQualifiedTypeName)
-            => new(fullyQualifiedTypeName, methodName: null, methodArity: 0, methodParameterCount: 0);
+        /// <summary>
+        /// Whether or not this is a strict search or not.  Strict searches require matching arity and parameter counts,
+        /// while non-strict searches do not.  Non-strict searches are useful in cases where the initial searching data
+        /// may not be produced in a well formed fashion (for example, some legacy test providers that do not follow:
+        /// https://github.com/microsoft/vstest-docs/blob/main/RFCs/0017-Managed-TestCase-Properties.md).
+        /// </summary>
+        [DataMember(Order = 4)]
+        public readonly bool Strict;
+
+        public static UnitTestingSearchQuery ForType(string fullyQualifiedTypeName, bool strict = true)
+            => new(fullyQualifiedTypeName, methodName: null, methodArity: 0, methodParameterCount: 0, strict);
 
-        public static UnitTestingSearchQuery ForMethod(string fullyQualifiedTypeName, string methodName, int methodArity, int methodParameterCount)
-            => new(fullyQualifiedTypeName, methodName, methodArity, methodParameterCount);
+        public static UnitTestingSearchQuery ForMethod(string fullyQualifiedTypeName, string methodName, int methodArity, int methodParameterCount, bool strict = true)
+            => new(fullyQualifiedTypeName, methodName, methodArity, methodParameterCount, strict);
 
-        private UnitTestingSearchQuery(string fullyQualifiedTypeName, string? methodName, int methodArity, int methodParameterCount)
+        private UnitTestingSearchQuery(string fullyQualifiedTypeName, string? methodName, int methodArity, int methodParameterCount, bool strict)
```

Comment:
One other thing we care about is presence of attributes on the method. Is it possible to introduce a flag to filter out methods that don't have any attributes? (Or perhaps hard code to exclude non-attributed methods when strict is true)

Reply:
yes.  i'll find a way to add those smarts in.